contracts/BIFI/strategies/Arbidex/StrategyArbidexMultiRewardsLP.sol

```diff
@@ -0,0 +1,338 @@
+        return IERC20(want).balanceOf(address(this));
+    }
+
+    // it calculates how much 'want' the strategy has working in the farm.
+    function balanceOfPool() public view returns (uint256) {
+        (uint256 _amount,,) = IMasterChef(chef).userInfo(poolId, address(this));
+        return _amount;
+    }
+
+    function rewardsAvailable() public view returns (uint256, uint256) {
+        uint256 arxAvailable = IMasterChef(chef).pendingArx(poolId, address(this));
+        uint256 wethAvailable = IMasterChef(chef).pendingWETH(poolId, address(this));
+        return (arxAvailable, wethAvailable);
+    }
+
+    function callReward() public view returns (uint256) {
+        IFeeConfig.FeeCategory memory fees = getFees();
+        (uint256 arxAvailable, uint256 wethAvailable) = rewardsAvailable();
+        uint256 nativeOut;
+       
+        if (arxAvailable > 0) {
+            uint256[] memory amountOut = IUniswapRouterETH(unirouter).getAmountsOut(arxAvailable, outputToNativeRoute);
+            nativeOut = nativeOut + amountOut[amountOut.length -1];
+        }
+       
+        if (wethAvailable > 0) {
+            nativeOut = nativeOut + wethAvailable;
+        }
+
+        return nativeOut * fees.total / DIVISOR * fees.call / DIVISOR;
+    }
+
+    function setHarvestOnDeposit(bool _harvestOnDeposit) external onlyManager {
+        harvestOnDeposit = _harvestOnDeposit;
+
+        if (harvestOnDeposit) {
+            setWithdrawalFee(0);
+        } else {
+            setWithdrawalFee(10);
+        }
+    }
+
+    function setShouldGasThrottle(bool _shouldGasThrottle) external onlyManager {
+        shouldGasThrottle = _shouldGasThrottle;
+    }
+
+    // called as part of strat migration. Sends all the available funds back to the vault.
+    function retireStrat() external {
+        require(msg.sender == vault, "!vault");
+
+        IMasterChef(chef).emergencyWithdraw(poolId);
+
+        uint256 wantBal = IERC20(want).balanceOf(address(this));
+        IERC20(want).transfer(vault, wantBal);
+    }
+
+    // pauses deposits and withdraws all funds from third party systems.
+    function panic() public onlyManager {
+        pause();
+        IMasterChef(chef).emergencyWithdraw(poolId);
+    }
+
+    function pause() public onlyManager {
+        _pause();
+
+        _removeAllowances();
+    }
+
+    function unpause() external onlyManager {
+        _unpause();
+
+        _giveAllowances();
+
+        deposit();
+    }
+
+    function _giveAllowances() internal {
+        IERC20(want).safeApprove(chef, type(uint).max);
+        IERC20(output).safeApprove(unirouter, type(uint).max);
+        IERC20(native).safeApprove(unirouter, type(uint).max);
+
+        IERC20(lpToken0).safeApprove(unirouter, 0);
+        IERC20(lpToken0).safeApprove(unirouter, type(uint).max);
+
+        IERC20(lpToken1).safeApprove(unirouter, 0);
+        IERC20(lpToken1).safeApprove(unirouter, type(uint).max);
+
+        if (rewardToNativeRoute.length != 0) {
```

Comment:
this `if` before `for` is probably not needed?

Reply:
Good catch, thanks